index.js

```diff
@@ -44,6 +45,44 @@ Counter.prototype.add = function(id, options){
 }
 
 /**
+ *    
+ * Note: Performance and memory usage could be greatly improved by:
+ * 1. removing the need for a newly allocated 'options' object everytime 'add' method is called
+ * 2. replacing 'options' with primitive types (Number, Boolean)
+ * This would lower the load on the GC during its garbage collection cycle, reduce heap memory footprint, and lower execution time
+ * 
+ * Adds an element to the counter, or increases
+ * its quantity if it was already in the counter.
+ * 
+ *
+ *
+ * Adding might fail if there is a reference that
+ * hold a lower count for the given ID. You can
+ * specify parameter force: true in the options to override
+ * this behaviour.
+ *
+ * @param  {String} id
+ * @param  {Number} qty  
+ * @param  {Boolean} force 
+ * @return {Boolean}         Whether adding has been succesfull or not
+ */
+Counter.prototype.addStrict = function(id, qty, force){
```

Comment:
You will very rarely need to loop through milions of records, so you'd probably prefer to use a more elegant api like `counter.add(...)`. But for the cases where performance is a real problem, I'm open to have a method for that -- can you rename it to `addFast(id, qty, force)` and add tests so that we're sure it works well? With those changes I'd be glad to merge! :)